### PR TITLE
Switch from Berksfile to Policyfile

### PR DESCRIPTION
**Pull request by damacus**
### Description

- Add a stub recipe because Policyfile blows up without it
- Add the Policyfile with the named_run_list from the Habitat coobkook
  (thanks @sam1el)
- Fix the spec_helper to use Policyfile

### Issues Resolved

Berkshelf doesn't currently support Ruby 3.0's splat, Policyfile does so we're going to switch. 

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by bmhughes** [comment]
I echo my comments from slack, I just don't see the benefit this gives 🤷‍♂️ but I'm pretty neutral feeling about it in general so I'm happy to go with the majority.

I'd switch the named run list out for something like below though to keep it all in block scope, not to have to worry about "." and ".." and it's more idiomatic ruby imo.

```ruby
Dir.each_child('./test/cookbooks/test/recipes') { |t| named_run_list test.to_sym, "test:#{test}" }
```

**Comment by bmhughes** [comment]
Ah I just seen that we're hitting a bug, so now it has a benefit ignore that part of ^.

File: Policyfile.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name 'haproxy'
+
+default_source :supermarket, 'https://supermarket.chef.io'
+cookbook 'haproxy', path: '.'
+cookbook 'test', path: 'test/cookbooks/test'
+
+run_list 'haproxy'
+
+tests = (Dir.entries('./test/cookbooks/test/recipes').select { |f| !File.directory? f })
```
Comment (bmhughes) [inline Policyfile.rb:9]: Can be replaced with ``Dir.each_child('./test/cookbooks/test/recipes') { |test| named_run_list test.to_sym, "test:#{test}" }``

**Review by bmhughes — DISMISSED** [review]

File: Policyfile.rb (hunk 1 above)
Comment (bmhughes) [reply Policyfile.rb:9]: Yes it can replace the lot, I'm not sure if there's a way to review multiple lines but that's what I meant.

Comment (bmhughes) [reply Policyfile.rb:9]: Yes it can replace the lot, I'm not sure if there's a way to review multiple lines but that's what I meant.

Comment (sam1el) [reply Policyfile.rb:9]: Cool! testing it locally on one of my cookbooks. i like a 1 liner much better for this

Comment (damacus) [reply Policyfile.rb]: I didn't find a nicer way to strip .rb off the end of the file so the test = test.gsub is still there

Comment (bmhughes) [reply Policyfile.rb]: `test` -> `File.basename(test, '.*')`

Thought I'd put that in there but apparently not.

Comment (bmhughes) [reply Policyfile.rb]:
```ruby
Dir.each_child('./test/fixtures/cookbooks/test/recipes') { |t| named_run_list File.basename(t, .*).to_sym, "test:#{File.basename(t, .*)}" }
```

Comment (sam1el) [reply Policyfile.rb]: finally got it working. was missing a few `:` tested in my other cookbooks
```ruby
Dir.each_child('./test/cookbooks/test/recipes') { |t| named_run_list :"#{File.basename(t, '.*')}".to_sym, "test::#{File.basename(t, '.*')}" }
```

Comment (bmhughes) [reply Policyfile.rb]: That's curious (assuming named_run_list is a method and not an attribute) as basename returns a string so the string interpolation should be superfluous.

Comment (sam1el) [reply Policyfile.rb]: the `named_run_list` name here wouldn't be a string though. it is more like creating a property in a cookbook resource. in that sense, `:this_recipe` is then consumed inside the `kitchen.yml` as `named_run_list: this_recipe`

**Review by Xorima — APPROVED** [review]

**Pull request closed without merging**